sno/diff.py

```diff
@@ -734,6 +738,33 @@ def _out(dataset, diff):
     yield _out
 
 
+def resolve_output_path(output_path):
+    if isinstance(output_path, io.IOBase):
+        return output_path
+    elif (not output_path) or output_path == "-":
+        return sys.stdout
+    else:
+        return output_path.open("w")
```

Comment:
Should we use a mode here?

Reply:
output_path can be a directory as well (eg: for geojson files)

Reply:
>  Should we use a mode here?

I assume you mean "a" for append-mode. I think probably not? For most formats (json, geojson, html) it doesn't make sense to append. I don't know why you'd want to for text either. if you need to you can just use `-` and then use `>> outfile` to do it in your shell

Reply:
>  output_path can be a directory as well (eg: for geojson files)

this function can't handle that, but `diff_output_geojson` deals with it specially

Reply:
> I assume you mean "a" for append-mode.

I meant text-mode with an encoding, since we're writing text?

Reply:
ah right. 

I don't think this is an issue nowadays? Everything defaults to UTF-8 in recent python versions, even if you have other locale settings:

```

LANG="en_NZ.latin-1" LC_COLLATE="en_NZ.latin-1" LC_CTYPE="en_NZ.latin-1" LC_MESSAGES="en_NZ.latin-1" LC_MONETARY="en_NZ.latin-1" LC_NUMERIC="en_NZ.latin-1" LC_TIME="en_NZ.latin-1" LC_ALL= ipython3

...

>>> import sys, locale, os
>>> locale.getpreferredencoding(True), sys.getdefaultencoding()
('UTF-8', 'utf-8')

```